China lake implementation

NotDeirun opened the pull request:
This implements a pump-action grenade launcher and new, projectile based grenade shells for it.
Currently only adds to the code, I will convert the other grenade launchers to the new projectile grenades later and then add the China Lake to the game itself.

NotDeirun commented:
Made a mistake with some of the dmi files, let me fix.

mattroks101 commented on code/modules/projectiles/projectile/projectilegrenades.dm:
```diff
@@ -0,0 +1,60 @@
+	var/heavy_emp_range = 3
+	var/light_emp_range = 8
+
+obj/item/projectile/bullet/grenade/emp/grenade_effect(target)
```
```suggestion
obj/item/projectile/bullet/grenade/emp/grenade_effect(target)
```

NotDeirun commented:
> 
> 
> This appears to have a lot of copy and paste code from pump shotguns. It might be best if this was just a child of pump shotguns that fired a grenade round, instead of copying and pasting shotgun code.

Since I plan to convert the other grenade launchers after this: Do you think making them all children of pump shotguns, making some of them children of pump shotguns or making them a child of some general grenade launcher would be the best idea?

mattroks101 commented:
All the ones that involve pumping should be children of shotguns, all the ones that have other special mechanics can be their own thing. Whatever the gun shares the most in common with, so if it behaves identical to a shotgun except it shoots a grenade, then make it a shotgun subtype.

NotDeirun commented:
Sounds like a good idea. I hope this works.

mattroks101 commented:
This PR has conflicts now.
![image](https://user-images.githubusercontent.com/6728268/77826276-d06ed400-70e4-11ea-9eb5-b436a282ec55.png)

NotDeirun commented:
Those were in from the start - I used old dmi files. I do think I fixed it though, if I take a look at the diffrences between the two files now I only see the sprites I added.

NotDeirun commented:
How can I resolve that conflict? The only changes I see, after I fixed my earlier mistake, are still just my additions.

MadmanMartian commented:
You'll need to

 - Extract your icon changes
 - Point towards eris's master branch, and pull to begin rebasing
 - When the DMI conflicts show up, select the Eris's master version to replace your own
 - Once the rebase is complete, re-add your icon changes
 - Point towards your remote's working branch (this branch) and push changes

NotDeirun commented:
I followed those steps until the last, but when trying to push to my remote I get this error message:
failed to push some refs to 'https://github.com/NotDeirun/CEV-Eris'
To https://github.com/NotDeirun/CEV-Eris
! refs/heads/China_Lake:refs/heads/China_Lake [rejected] (non-fast-forward)
Done
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
'China_Lake' rejected (non-fast-forward)
error: failed to push some refs to 'https://github.com/NotDeirun/CEV-Eris'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

MadmanMartian commented:
You'll need to force push your changes

NotDeirun commented:
Thanks for the help, this almost drove me insane.

MadmanMartian commented on cev_eris.dme:
```diff
@@ -2185,6 +2185,7 @@
 #include "code\modules\projectiles\guns\launcher\crossbow.dm"
 #include "code\modules\projectiles\guns\launcher\grenade_launcher.dm"
 #include "code\modules\projectiles\guns\launcher\pneumatic.dm"
+#include "code\modules\projectiles\guns\launcher\projectile_greande_launcher.dm"
```
Typo in the filename

NotDeirun replied:
Fixed.
